### PR TITLE
Enable nullability in DragAssistanceManager

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in DragAssistanceManager.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/10749)

### CODE REVIEW

**Review by LeafShi1 — COMMENTED** [review]
Looks good!

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/DragAssistanceManager.Line.cs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                return new Line(Math.Min(l1.x1, l2.x1), l1.y1, Math.Max(l1.x2, l2.x2), l1.y2, l1.LineType);
+            }
+
+            return null;
+        }
+
+        public override string ToString()
+        {
+            return $"Line, type = {_lineType}, dims =({x1}, {y1})->({x2}, {y2})";
+        }
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/DragAssistanceManager.Line.cs:111]: please use body expression

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/DragAssistanceManager.Line.cs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        {
+            get => _originalBounds;
+            set => _originalBounds = value;
+        }
+
+        public PaddingLineType PaddingLineType
+        {
+            get => _paddingLineType;
+            set => _paddingLineType = value;
+        }
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/DragAssistanceManager.Line.cs:36]: Do we need backing fields for these? Let's make this into auto properties if we can.

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/DragAssistanceManager.cs [hunk 3]
```diff
@@ -460,11 +501,19 @@ private void Initialize(List<IComponent> dragComponents, IDesignerHost host)
                     disposeDesigner = true;
                 }
             }
+            else
+            {
+                designer = controlDesigner;
+            }
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/DragAssistanceManager.cs:507]: why was this added?

Comment (gpetrou) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/DragAssistanceManager.cs:507]: Because designer needs to be set.

**Review by lonitra — APPROVED** [review]
👍
